### PR TITLE
refactor(examples): add rendezvous point fixed keypair

### PR DESCRIPTION
**Pull request by bgins**
## Description

Related: #4380.

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation (No changes needed)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Comment by bgins** [comment]
I was a bit terse about the checklist because it ends up the commit. Let me know docs, changelog entries, or tests should be added.

File: examples/rendezvous/src/main.rs [hunk 1]
```diff
@@ -33,7 +33,9 @@ use std::time::Duration;
 async fn main() -> Result<(), Box<dyn Error>> {
     env_logger::init();
 
-    let mut swarm = libp2p::SwarmBuilder::with_new_identity()
+    let keypair = libp2p::identity::Keypair::ed25519_from_bytes([0; 32]).unwrap();
```
Comment (thomaseizinger) [inline examples/rendezvous/src/main.rs:38]: Can we perhaps add a comment here that says something like: "Results in `PeerId` XYZ which is what we hardcode in the other examples"?

**Comment by thomaseizinger** [comment]
> I was a bit terse about the checklist because it ends up the commit. Let me know docs, changelog entries, or tests should be added.

I've updated the description for you, thanks! :)

**Review by thomaseizinger — APPROVED** [review]
Thanks for picking this up again!

One suggestion, otherwise good to merge :)

PS: Are you interested in adding https://github.com/ipvm-wg/homestar to https://github.com/libp2p/rust-libp2p?tab=readme-ov-file#notable-users?

File: examples/rendezvous/src/main.rs (hunk 1 above)
Comment (bgins) [reply examples/rendezvous/src/main.rs:38]: Yep! Added a comment. :)

**Comment by bgins** [comment]
> PS: Are you interested in adding https://github.com/ipvm-wg/homestar to https://github.com/libp2p/rust-libp2p?tab=readme-ov-file#notable-users?

Yes absolutely! Here it is: https://github.com/libp2p/rust-libp2p/pull/4669

**Comment by bgins** [comment]
Closed in favor of #4681

**Pull request closed without merging**